orbit/cmd/orbit/orbit.go

```diff
@@ -904,9 +876,16 @@ func main() {
 			default:
 				logging.LogErrIfEnvNotSet(constant.SilenceEnrollLogErrorEnvVar, err, "error with extensions.load file at "+extensionAutoLoadFile)
 			}
-			g.Add(extRunner.Execute, extRunner.Interrupt)
+
+			orbitClient.RegisterConfigReceiver(extRunner)
```

Comment:
Just wondering out loud, are we losing any important feature by moving away from the `g.Add` behavior? (we were doing finer-grained groups of `g.Add` vs one to run all ConfigReceivers IIUC?)

Reply:
My understanding is that the application framework we use will close the entire application if any of the Execute functions return. That is currently used to force orbit to restart when there is a change of launch flags or an application update. It also lets the config loop run on an independent timer, although in almost all cases they have the same wait time as the main config fetcher. I could be wrong about the motivation for that choice but that seems to be the effect.

With this change, we would instead have a cancel func that you can pass into receivers that may at some point wish to restart the application. One of the benefits of that is that now all restarts are explicit function calls, instead of being implicit on the return of a function. It was a little confusing to understand what was happening at first because that expected behaviour wasn't documented anywhere

Reply:
> That is currently used to force orbit to restart when there is a change of launch flags or an application update. It also lets the config loop run on an independent timer, although in almost all cases they have the same wait time as the main config fetcher.

That is also my understanding. That's also something Lucas could probably give more context about, to make sure the new approach covers all expected behaviors.

Reply:
the change also affects those loops (config fetcher with middlewares, flags fetcher and server overrides runner) in the "opposite direction", meaning that they can't interrupt each other anymore (they're in the same `RunConfigReceivers` loop now)

IMO this is good, but yeah I don't know if there was an implicit behavior depending on that.